pkgs/applications/networking/cluster/temporal-cli/default.nix

```diff
@@ -1,35 +1,83 @@
-{ lib, fetchFromGitHub, buildGoModule, testers, temporal-cli }:
+{ lib, fetchFromGitHub, buildGoModule, installShellFiles, symlinkJoin }:
 
-buildGoModule rec {
-  pname = "temporal-cli";
-  version = "1.18.0";
+let
+  tctl-next = buildGoModule rec {
+    pname = "tctl-next";
+    version = "0.8.0";
 
-  src = fetchFromGitHub {
-    owner = "temporalio";
-    repo = "tctl";
-    rev = "v${version}";
-    hash = "sha256-LcBKkx3mcDOrGT6yJx98CSgxbwskqGPWqOzHWOu6cig=";
+    src = fetchFromGitHub {
+      owner = "temporalio";
+      repo = "cli";
+      rev = "v${version}";
+      hash = "sha256-yQnFw3uYGKrTevGFVZNgkWwKCCWiGy0qwJJOmnMpTJQ=";
+    };
+
+    vendorHash = "sha256-ld59ADHnlgsCA2mzVhdq6Vb2aa9rApvFxs3NpHiCKxo=";
+
+    nativeBuildInputs = [ installShellFiles ];
+
+    excludedPackages = [ "./cmd/docgen" ];
+
+    ldflags = [
+      "-s"
+      "-w"
+      "-X github.com/temporalio/cli/headers.Version=${version}"
+    ];
+
+    preCheck = ''
+      export HOME=$(mktemp -d)
+    '';
+
+    postInstall = ''
+      installShellCompletion --cmd temporal \
+        --bash <($out/bin/temporal completion bash) \
+        --zsh <($out/bin/temporal completion zsh)
+    '';
   };
 
-  vendorHash = "sha256-BUYEeC5zli++OxVFgECJGqJkbDwglLppSxgo+4AqOb0=";
+  tctl = buildGoModule rec {
+    pname = "tctl";
+    version = "1.18.0";
 
-  ldflags = [ "-s" "-w" ];
+    src = fetchFromGitHub {
+      owner = "temporalio";
+      repo = "tctl";
+      rev = "v${version}";
+      hash = "sha256-LcBKkx3mcDOrGT6yJx98CSgxbwskqGPWqOzHWOu6cig=";
+    };
 
-  preCheck = ''
-    export HOME=$(mktemp -d)
-  '';
+    vendorHash = "sha256-BUYEeC5zli++OxVFgECJGqJkbDwglLppSxgo+4AqOb0=";
 
-  passthru.tests.version = testers.testVersion {
-    package = temporal-cli;
-    # the app writes a default config file at startup
-    command = "HOME=$(mktemp -d) ${meta.mainProgram} --version";
+    nativeBuildInputs = [ installShellFiles ];
+
+    excludedPackages = [ "./cmd/copyright" ];
+
+    ldflags = [ "-s" "-w" ];
+
+    preCheck = ''
+      export HOME=$(mktemp -d)
+    '';
+
+    postInstall = ''
+      installShellCompletion --cmd tctl \
+        --bash <($out/bin/tctl completion bash) \
+        --zsh <($out/bin/tctl completion zsh)
+    '';
   };
+in
+symlinkJoin {
+  name = "temporal-cli-${tctl.version}";
```

Comment:
```suggestion
  pname = "temporal-cli";
  inherit (tctl) version;
  name = "${pname}-${version}";
```

Reply:
Isn't `name` enough for `symlinkJoin`?

Reply:
Yes but if we are treating temporal-cli like a package then it is expected that it has pname+version.

Reply:
Done.